source/type/data.cpp

```diff
@@ -413,6 +414,48 @@ void Data::aggregate_odt(){
     for(JourneyPattern* jp : this->pt_data->journey_patterns){
         jp->build_odt_properties();
     }
+
+    // cf http://confluence.canaltp.fr/pages/viewpage.action?pageId=3147700 (we really should put that public)
+    // for some ODT kind, we have to fill the Admin structure with the ODT stop points
+    std::unordered_map<georef::Admin*, std::set<const nt::StopPoint*>> odt_stops_by_admin;
+    for (const auto jp: pt_data->journey_patterns) {
+        if (! jp->odt_properties.is_zonal_odt()) {
+            continue;
+        }
+        if (jp->journey_pattern_point_list.size() != 2) {
+            LOG4CPLUS_WARN(log4cplus::Logger::getInstance("log"), "it's strange, a zone odt journey pattern ("
+                           << jp->uri << ") has more than 2 stops, we skip it");
+            continue;
+        }
+        bool add = false;
+        // we add it for the ODT type where the vehicle comes directly to the user
+        jp->for_each_vehicle_journey([&](const VehicleJourney& vj) {
+            if (in(vj.vehicle_journey_type, {VehicleJourneyType::adress_to_stop_point,
+                     VehicleJourneyType::odt_point_to_point} )) {
+                add = true;
+                return false; // we can stop
+            }
+            return true;
+        });
+
+        if (! add ) {
+            continue;
+        }
+
+        for (const auto jpp: jp->journey_pattern_point_list) {
+            const auto sp = jpp->stop_point;
+            for (auto* admin: sp->admin_list) {
+                odt_stops_by_admin[admin].insert(sp);
+            }
+        }
+    }
+
+    //we first store the stops in a set not to have dupplicates
+    for (const auto& p: odt_stops_by_admin) {
+        for (const auto& sp: p.second) {
+            p.first->odt_stop_points.push_back(sp);
```

Comment:
I don't think so, It is only used  to have free stoppoints on those admin, the order is irrelevant.